### DISCUSSION TITLE
Question: [VirtualizingContext.RealizationRect bounds]

### DISCUSSION THREAD

**Discussion by BorzillaR** [microsoft/microsoft-ui-xaml#7510] [Q&A]
Hello!

I try to checkout ItemsRepeater for new WinUI (Desktop). I want to write a simple list of strings with a virtualization UI. To start with I copy ActivityFeedLayout from Xaml Controls Gallery sample. Next, I modify it that it shows only one column. As ItemTemplate I use a TextBlock. When I set VerticalCacheLength and HorizontalCacheLength  to 0 to see how items look like, I notice that the layout does not start from first item. Instead of this there is some blank space at the top of the ScrollViewer. 

![image](https://user-images.githubusercontent.com/23479280/94143249-3d86da80-fe78-11ea-82fe-b151965ba58b.png)

To know what happens I debug the code and see that RealizationRect.Y which comes on measure is not zero. Could you explane why this happens and help me to understand and fix this issue?

My project is attached bellow

[ItemsRepeater.zip](https://github.com/microsoft/microsoft-ui-xaml/files/5275773/ItemsRepeater.zip)

**Comment by StephenLPeters** [comment]
@ranjeshj can you weigh in?

**Comment by ranjeshj** [comment]
This looks like the issue with effective viewport not providing the correct bounds. Boiled down repro steps are available in this [issue](https://github.com/microsoft/microsoft-ui-xaml/issues/3001).

**Comment by BorzillaR** [comment]
This issue still exists in Preview 4
[Updated project](https://github.com/microsoft/microsoft-ui-xaml/files/5982239/App24.zip)
